crates/holochain_p2p/tests/tests/node_messaging.rs

```diff
@@ -171,6 +206,183 @@ async fn test_remote_signal() {
+            before: None,
+            author: None,
+        },
+        holochain_p2p::actor::GetLinksOptions::default(),
+    )
+    .await
+    .unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_count_links() {
+    let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
+    let space = dna_hash.to_k2_space();
+    let handler = Arc::new(Handler::default());
+
+    let (_agent1, hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (_agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+
+    hc1.test_set_full_arcs(space.clone()).await;
+    hc2.test_set_full_arcs(space.clone()).await;
+
+    // give some time for the full arcs to propagate
+    tokio::time::sleep(std::time::Duration::from_millis(20)).await;
+
+    // if we get a response at all, the full back-n-forth succeeded
+    hc2.count_links(
+        dna_hash,
+        WireLinkQuery {
+            base: HoloHash::from_raw_36_and_type(vec![1; 36], holo_hash::hash_type::AnyDht::Entry)
+                .into(),
+            link_type: LinkTypeFilter::Types(Vec::new()),
+            tag_prefix: None,
+            before: None,
+            after: None,
+            author: None,
+        },
+    )
+    .await
+    .unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_get_agent_activity() {
+    let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
+    let space = dna_hash.to_k2_space();
+    let handler = Arc::new(Handler::default());
+
+    let (_agent1, hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (_agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+
+    hc1.test_set_full_arcs(space.clone()).await;
+    hc2.test_set_full_arcs(space.clone()).await;
+
+    // give some time for the full arcs to propagate
+    tokio::time::sleep(std::time::Duration::from_millis(20)).await;
+
+    // if we get a response at all, the full back-n-forth succeeded
+    hc2.get_agent_activity(
+        dna_hash,
+        AgentPubKey::from_raw_36(vec![2; 36]),
+        ChainQueryFilter {
+            sequence_range: ChainQueryFilterRange::Unbounded,
+            entry_type: None,
+            entry_hashes: None,
+            action_type: None,
+            include_entries: false,
+            order_descending: false,
+        },
+        holochain_p2p::actor::GetActivityOptions::default(),
+    )
+    .await
+    .unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_must_get_agent_activity() {
+    let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
+    let space = dna_hash.to_k2_space();
+    let handler = Arc::new(Handler::default());
+
+    let (_agent1, hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (_agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+
+    hc1.test_set_full_arcs(space.clone()).await;
+    hc2.test_set_full_arcs(space.clone()).await;
+
+    // give some time for the full arcs to propagate
+    tokio::time::sleep(std::time::Duration::from_millis(20)).await;
```

Comment:
I can just feel myself pressing "re-run" on Windows. If we can possibly configure K2 to operate fast enough that it's practical to do a "retry with timeout" pattern here, I'd prefer to do that now

Reply:
Done: https://github.com/holochain/holochain/pull/4743/commits/69b986300d23e5ec00c25bd1b7ab9b0ddd732679